datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -103,6 +103,7 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
         # Update the default behavior for global settings
         default_fields['log_requests'] = init_config.get('log_requests', default_fields['log_requests'])
         default_fields['skip_proxy'] = init_config.get('skip_proxy', default_fields['skip_proxy'])
+        default_fields['timeout'] = init_config.get('timeout', default_fields['timeout'])
```

Comment:
Would that make sense to also add `connect_timeout` and `read_ timeout` to init config ?

Reply:
This will be addressed in a later PR that would set default_fields for all config options to the `init_config` value if it exists.